core/primitives/src/types.rs

```diff
@@ -34,7 +34,7 @@ pub type PromiseId = Vec<ReceiptIndex>;
 #[derive(Hash, Eq, PartialEq, Clone, Debug, BorshSerialize, BorshDeserialize, Default)]
 pub struct StateRoot {
     pub hash: CryptoHash,
-    pub num_parts: u64,
+    pub state_size: u64,
```

Comment:
What is this state_size? Please add comments and write a highlevel doc.

Reply:
/// `state_size` is the size of the state in bytes, identified by StateRoot `hash`

This field is important when we want to decide of how many parts we want to split and request the state. The point why it's here is the same as for `num_parts`: we should prove to rely on valid state size while splitting / collecting the state.
Could you please explain what is highlevel doc? I can add comments to `pub struct StateRoot` description. Can I do something else reasonable?

Reply:
Discussed with @mikhailOK offline. He said, it's possible to get and prove the node with hash `hash` easily. The node already contains of memory usage.
Ignore the comment above.